frontend/src/features/homes/services/homes-list.js

```diff
@@ -0,0 +1,72 @@
+import { $axios } from "src/boot/axios";
+import { Notify } from "quasar";
+import { errorNotifier } from "src/utils/notifier.js";
+import { i18n } from "src/boot/i18n";
+
+export const getUserGroups = async () => {
+  try {
+    const { data } = await $axios.post("/methods/currentUserGroups");
+    return data;
+  } catch (error) {
+    errorNotifier(error);
+    return [];
+  }
+};
+
+export const getHomesWithActivityLevel = async groupId => {
+  try {
+    const { data } = await $axios.post("/methods/getHomesWithActivityLevel", {
+      groupId
+    });
+    return data;
+  } catch (error) {
+    errorNotifier(error);
+    return [];
+  }
+};
+
+export const addOrUpdateAGroup = async groupDetails => {
+  try {
+    let payload = groupDetails;
+    if (groupDetails._id) {
+      payload = { _id: groupDetails._id, modifier: { $set: groupDetails } };
+      delete payload.modifier._id;
+    }
+    await $axios.post("/methods/addOrUpdateAGroup", payload);
+
+    Notify.create({
+      position: "top-right",
+      type: "positive",
+      message: i18n.t(
+        `groupModal-${groupDetails._id ? "update" : "create"}-successful`
+      )
+    });
+    return true;
+  } catch (error) {
+    errorNotifier(error);
+    return false;
+  }
+};
+
+export const addOrUpdateHome =async homeDetails => {
+  try {
+    let payload = homeDetails;
+    if (homeDetails._id) {
+      payload = { _id: homeDetails._id, modifier: { $set: homeDetails } };
+      delete payload.modifier._id;
+    }
+    await $axios.post("/methods/addOrUpdateHome", payload);
+
+    Notify.create({
+      position: "top-right",
+      type: "positive",
+      message: i18n.t(
+        `homeModal-${homeDetails._id ? "update" : "create"}-successful`
+      )
+    });
```

Comment:
Consider moving this to a helper function, so it can be handled outside of the service.